### PR TITLE
Fix Android build in CI

### PR DESCRIPTION
**Pull request by thunderbiscuit**
Not yet sure why the Android build breaks. Might have to do with the update to the ubuntu-22.04 image.

### CODE REVIEW

**Comment by yellowHatpro** [comment]
I might be super wrong, but isn't ubuntu 20.04 also affected due to NDK replacement?

**Comment by thunderbiscuit** [comment]
Yep indeed they both need the manual download of NDK 21. I wondered whether there was something else about the ubuntu-22.04 image that might have caused the issue, but indeed both versions return the same error. So I don't think the upgrade to the 22.04 version was the problem in the end.

**Comment by notmandatory** [comment]
Give this a try to fix the build: https://github.com/bitcoindevkit/bdk-ffi/pull/348/commits/afa284f0b9d45b78de1a973f8851e6cf7e62eb34

I think it broke because of an issue in rust 1.68 which is default in ubuntu 20.04 and 22.04. Sounds like this issue: https://github.com/cross-rs/cross/issues/1222

**Comment by thunderbiscuit** [comment]
I have added a workflow_dispatch to all test workflows so as to be able to trigger them manually in the future (so as to not need the 1 line changes in the readmes just to run specific workflows).

File: bdk-jvm/README.md [hunk 1]
```diff
@@ -98,3 +98,5 @@ dependencies {
 
 [`bdk`]: https://github.com/bitcoindevkit/bdk
 [`bdk-ffi`]: https://github.com/bitcoindevkit/bdk-ffi
+
+Temporary line: needed to run CI
```
Comment (notmandatory) [inline bdk-jvm/README.md:102]: Did you forgot to remove this line?

File: bdk-python/README.md [hunk 2]
```diff
@@ -46,3 +46,5 @@ tox -vv
 ```shell
 pip install ./dist/bdkpython-<yourversion>-py3-none-any.whl
 ```
+
+Temporary line: needed to run CI
```
Comment (notmandatory) [inline bdk-python/README.md:50]: Did you forgot to remove this line?

File: bdk-swift/README.md [hunk 3]
```diff
@@ -50,3 +50,5 @@ own GitHub repository use the following steps:
 [Xcode]: https://developer.apple.com/documentation/Xcode
 [`bdk`]: https://github.com/bitcoindevkit/bdk
 [`bdk-ffi`]: https://github.com/bitcoindevkit/bdk-ffi
+
+Temporary line: needed to run CI
```
Comment (notmandatory) [inline bdk-swift/README.md:54]: Did you forgot to remove this line?

File: .github/workflows/publish-jvm.yaml [hunk 4]
```diff
@@ -43,7 +46,7 @@ jobs:
   build-jvm-full-library:
     name: Create full bdk-jvm library
     needs: [build-jvm-macOS-M1-native-lib]
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-20.04
```
Comment (notmandatory) [inline .github/workflows/publish-jvm.yaml:49]: Should the `publish-android`, `publish-python` and `test-python` or any other places with `ubuntu-22.04` also be changed to `ubuntu-20.04`?

**Review by notmandatory — COMMENTED** [review]
ACK 52b6dfb9ff901dd6471f0d244a43a7653b4fa8b1, with a few comments.

**Comment by thunderbiscuit** [comment]
Sorry man I left this PR not in its final state this weekend. Today I'll squash all commits into one, and remove the extra lines in the readmes (I still needed them yesterday to make sure I ran all the workflows with the latest changes on the yaml, but now that they have run and succeeded, I can remove them).

The idea of running everything on 20.04 I think might be a good one for the short term. I'll fix those everywhere as well.

File: bdk-swift/README.md (hunk 3 above)
Comment (thunderbiscuit) [reply bdk-swift/README.md:54]: I had a few more changes to the yaml files so I still needed them yesterday for my last push, but now that everything has passed I will remove them.

File: bdk-python/README.md (hunk 2 above)
Comment (thunderbiscuit) [reply bdk-python/README.md:50]: Will remove now.

**Comment by thunderbiscuit** [comment]
The build for the older versions of Python breaks, but I'm also realizing that we are supporting probably way too old versions; Python 3.6 is 8 years old at this point, and even Debian stable (version 10, Bullseye) ships with 3.9. I feel it fair to expect users of the bitcoin development kit Python bindings use a recent-ish version of Python, and propose we deprecate support for 3.6 and 3.7.

**Comment by thunderbiscuit** [comment]
Note that I also removed the PyPy releases in favour of just the CPython ones. I'd be happy to bring them back if we get demand for it, but at this point I don't know that it's really useful, and the versions it was even supporting were 3.7 and 3.8.

I think the cleaner approach is to start by supporting recent versions of CPython, and address demands for older/esoteric versions if and as they come.

**Review by notmandatory — APPROVED** [review]
ACK 871a06d1ce39456a2283cca5a927379a10db22ef

Nice well organized PR!

**Pull request merged**